main/src/main/java/cgeo/geocaching/storage/DataStore.java

```diff
@@ -2339,6 +2339,31 @@ public static void saveUserModifiedCoords(final Geocache cache) {
         });
     }
 
+    public static boolean saveFinalDefinedStatus(final Geocache cache) {
+        cache.addStorageLocation(StorageLocation.DATABASE);
+        cacheCache.putCacheInCache(cache);
+        Log.d("Updating finalDefined of " + cache + " in DB");
+
+        final ContentValues values = new ContentValues();
+        values.put("finalDefined", cache.hasFinalDefined() ? 1 : 0);
+
+        init();
```

Comment:
Surround all database access with a call to `withAccessLock` (see how other methods in Data store are doing it)

Reply:
Merged the new function into saveWaypoint